libs/include/geometry/types-2d.h

```diff
@@ -2,25 +2,41 @@
 
 #include <vector>
 
+#include "constants.h"
+
+struct Cartesian2d {
+  /// @brief Coordinate in x axis.
+  float x = 0;
+  /// @brief Coordinate in y axis.
+  float y = 0;
+};
+
+struct Circular2d {
+  /// @brief Distance of point to origin.
+  float m = 0;
+  /// @brief Angle between x axis and line segment to origin.
+  float a = 0;
+};
+
 /// @brief Two Dimensional Point.
 struct Point2d {
-  float x;
-  float y;
+  Cartesian2d ca = Cartesian2d{};
+  Circular2d ci = Circular2d{};
```

Comment:
I don't like this, I need a way to access the cartesian and circular more directly.
Problem is that most operations use cartesian coordinates and it's wasteful to keep updating the circular coordinates if they are not used.